OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs

```diff
@@ -405,6 +405,20 @@ public void OnInfiltrated(Actor a, LuaFunction func)
 			GetScriptTriggers(a).RegisterCallback(Trigger.OnInfiltrated, func, Context);
 		}
 
+		[Desc("Call a function when this actor is discovered by an enemy (doesn't work for neutral actors). " +
+			"The callback function will be called as func(Actor self, Player discoverer)")]
+		public void OnDiscovered(Actor a, LuaFunction func)
+		{
+			GetScriptTriggers(a).RegisterCallback(Trigger.OnDiscovered, func, Context);
+		}
+
+		[Desc("Call a function when this player is discovered by an enemy (doesn't work for neutrals). " +
+			"The callback function will be called as func(Actor self, Player discoverer)")]
```

Comment:
You should mention that you're returning the playeractor here.
Btw I'm not a fan of returning the playeractor. Don't we have another solution?
I'd still like to see ```func(Player discovered, Player discoverer)```.